example/lib/interactive_features_example.dart

```diff
@@ -0,0 +1,92 @@
+import 'package:flutter/material.dart';
+import 'package:mapbox_maps_flutter/mapbox_maps_flutter.dart';
+import 'example.dart';
+
+class InteractiveFeaturesExample extends StatefulWidget implements Example {
+  @override
+  final Widget leading = const Icon(Icons.map);
+  @override
+  final String title = 'Interactive Features';
+  @override
+  final String? subtitle = 'Click to select buildings, long-click to unselect';
+
+  @override
+  State<StatefulWidget> createState() => InteractiveFeaturesState();
+}
+
+class InteractiveFeaturesState extends State<InteractiveFeaturesExample> {
+  InteractiveFeaturesState();
+  MapboxMap? mapboxMap;
+
+  _onMapCreated(MapboxMap mapboxMap) {
+    this.mapboxMap = mapboxMap;
+    mapboxMap.style;
+  }
+
+  _onTap(context) async {
```

Comment:
For me it looks a bit strange that we're advertising the following approach of building highlight on a click. Honestly for me it looks more of an anti-approach comparing with Android / iOS 😄 The main reason for this is that in this example user defines their own order of click events, e.g.:

1. There's a POI on top of the building.
2. User clicks the POI.
3. User takes the code from this example and first implements `queryRenderedFeaturesForFeatureset` for buildings and then for POIs.
4. Then although actually POI was clicked - based on this code the building will be highlighted first.

I've just came from implementing interactions in Dash and it took me quite a while to re-write their logic with QRFs to interactions.

Bottom line: if we do not have time to add `addInteraction` - I'd at least modify this example and remove `onTap` at all to showcase highlighting the building for given coordinate. In the future Flutter versions we could prepare the proper example like:

```
_onTap(context) async {
 // map clicked
}

_onPoiTap(poi, context) async {
  // e.g. hide POI
}

_onBuildingTap(building, context) async {
  // highlight building
}
```

cc @persidskiy

Reply:
That's a fair point -- the example should more clearly demonstrate the logic we'd like the developer to think in. I can rewrite this today.

Reply:
Yeah, this PR implements the low-level API - QRF, but the most useful now is the `addInteraction()`.

Ideally, we would expose it here too, and used it in the example